File: LeetCodeProblemsCpp/LeetCodeProblemsCpp/Problems/LeetCode2419LongestSubarrayWithMaximumBitwiseAND.cpp
#include "LeetCode2419LongestSubarrayWithMaximumBitwiseAND.h"
using namespace std;

int LeetCode2419LongestSubarrayWithMaximumBitwiseAND::longestSubarray(vector<int>& nums)
{
    int max{};
    for (const auto& num : nums)
    {
      max= std::max(max,num);
    }
    int consecutiveCount{};
    int maxConsecutiveCount{};
    for (const auto& num : nums)
    {
        if (num==max)
        {
            consecutiveCount++;
            continue;
        }
        maxConsecutiveCount = std::max(maxConsecutiveCount, consecutiveCount);
        consecutiveCount = 0;
    }
    maxConsecutiveCount = std::max(maxConsecutiveCount, consecutiveCount);
    return  maxConsecutiveCount;
}

File: LeetCodeProblemsGo/LeetCode2419LongestSubarrayWithMaximumBitwiseAND/LeetCode2419LongestSubarrayWithMaximumBitwiseAND.go
package LeetCode2419LongestSubarrayWithMaximumBitwiseAND

func longestSubarray(nums []int) int {
	maxNum := 0
	for _, num := range nums {
		maxNum = max(maxNum, num)
	}
	consecutiveCount := 0
	maxConsecutiveCount := 0
	for _, num := range nums {
		if num == maxNum {
			consecutiveCount++
			continue
		}
		maxConsecutiveCount = max(maxConsecutiveCount, consecutiveCount)
		consecutiveCount = 0
	}
	maxConsecutiveCount = max(maxConsecutiveCount, consecutiveCount)
	return maxConsecutiveCount
}
